### ISSUE TITLE
Next release: 6.0.1 or 6.1.0?

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#2444]
We planned to have a release in February, but haven't decided it's a bug-fix release (6.0.1) or a minor release (6.1.0).

I prefer a minor release (6.1.0) with new features included. If that's the plan, then we probably should abandon the 6.0 branch and commit all fixes to the master branch.

**Comment by PaulWessel** [comment]
I guess I was the only one holding out for a 6.0.1 release but I think the rest prefer 6.1.0 - so that is what we will do.

**Comment by seisman** [comment]
So if everyone @joa-quim @remkos agree with this, then we should stop pushing to the 6.0 branch.

**Comment by joa-quim** [comment]
I've been saying this for a month.

**Comment by PaulWessel** [comment]
Can we block commits to the 6.0 branch in case we forget?

**Comment by seisman** [comment]
I just changed the settings of the 6.0 branch. I think now no one can push directly to this branch, but we still can open a pull request targeting the 6.0 branch.

**Comment by PaulWessel** [comment]
Building master and running tests I suddenly am getting 45 failures.  Is it just my system again or did something major change?

**Comment by seisman** [comment]
The CI jobs last night looks normal. The Linux tests fail due to connection timeout to the gmt data server.

**Comment by PaulWessel** [comment]
Strange, I am getting lots of these:

```
109/844 Testing: GMT_robinson.sh
Command: "/opt/local/bin/bash" "gmtest" "GMT_robinson.sh"
----------------------------------------------------------
PSL:../src/postscriptlight.c:5793: PSL_fopen error: Unable to open file /Users/pwessel/.gmt/sessions/gmt6.1811/gmt_0.ps- with mode w!
coast [ERROR]: Cannot open /Users/pwessel/.gmt/sessions/gmt6.1811/gmt_0.ps- with mode w
```

but ctest -R GMT_robinson.sh passes.
I have plenty of disc space.

**Comment by PaulWessel** [comment]
Also psconvert [ERROR]: gmtinit_set_last_dimensions: Could not create file /Users/pwessel/.gmt/sessions/gmt6.1811/gmt.canvas.0 for figure 0

Xcode 11.3 installed new command line tools this morning....

**Comment by PaulWessel** [comment]
Actually the full message was

psconvert [ERROR]: Cannot open file /Users/pwessel/.gmt/sessions/gmt6.1811/gmt_0.ps-
psconvert [ERROR]: gmtinit_set_last_dimensions: Could not create file /Users/pwessel/.gmt/sessions/gmt6.1811/gmt.canvas.0 for figure 0
end [ERROR]: No directory named /Users/pwessel/.gmt/sessions/gmt6.1811

which says the session directory for this job is not there, as if the PPID numbers are changing or not stable (again)?

**Comment by seisman** [comment]
I just ran all tests locally. 100% pass.

**Comment by PaulWessel** [comment]
After a full reboot, all is well.  Must be something odd related to the 11.3 update.

**Comment by seisman** [comment]
Closing.

**Comment by remkos** [comment]
You guys are fast. Did not have time to respond. But I agree with the idea of dropping `6.0`.
It wasn't very practical to keep making bug fixes to `6.0` and then having to merge with `master` again. And as we are not introducing any major changes that would hurt 6.0.0 users, 6.1.0 is the way to go.

I'd like to reopen this to have a clearer idea to what the next step is, so we don't get into this type of discussion again. My suggestion is:
* Keep developing in `master` through PRs.
* Do not create a `6.1` branch.
* If you are developing anything ground-braking, like the long options and the CPT hinges (to name a few), this is to be developed in a separate branch. These can only be merged through a PR and giving some of us the time to check not only the code, but also running it.
* As soon as a significant new feature is introduced we will know that the future version will have a minor version increase. So likely next version is 6.2.0. If not, then we will have 6.1.1.

**Comment by remkos** [comment]
@seisman : other than blocking any pushes to `6.0` you can now simply delete that branch on the GitHub site.

**Comment by seisman** [comment]
> @seisman : other than blocking any pushes to `6.0` you can now simply delete that branch on the GitHub site.

Will we lose something if we delete the 6.0 branch? e.g. some tags/releases are created from the 6.0 branch.

**Comment by seisman** [comment]
It seems safe to delete the 6.0 branch. I just deleted it from GitHub. Let's see if anything unexpected happens. Don't worry. We still have the 6.0 branch locally and can restore it in seconds.

**Comment by joa-quim** [comment]
Why do we want to delete the 6.0 branch? Aren't we loosing the commit messages made in 6.0 and that later show up in master as *merged from 6.0*?

**Comment by seisman** [comment]
Since all the commits are merged into master, deleting 6.0 branch doesn't lose anything.

However, we still have a few PRs which target for 6.0 branch. Deleting 6.0 branch automatically closed these PRs. So I have to revert the 6.0 branch and then change the base branch of these PRs from 6.0 to master.

**Comment by seisman** [comment]
I've updated the base branch of all the related PRs to master, and deleted 6.0 branch again. Please let me know if anything else is lost.

**Comment by remkos** [comment]
Regarding @joa-quim's question about losing history: no we don't. Yes, you'll see the "merged from `6.0`" but the commits merged also show up.
Sound like we can close this. Thanks @seisman for doing the legwork.
